Tests/Listener/RestListenerLocaleTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Paysera\Bundle\RestBundle\Tests;
+
+use Mockery;
+use Psr\Log\NullLogger;
+use Mockery\MockInterface;
+use PHPUnit_Framework_TestCase;
+use Paysera\Bundle\RestBundle\ApiManager;
+use Symfony\Component\HttpFoundation\Request;
+use Paysera\Bundle\RestBundle\Service\RequestLogger;
+use Paysera\Bundle\RestBundle\Listener\RestListener;
+use Paysera\Bundle\RestBundle\Service\ExceptionLogger;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Paysera\Bundle\RestBundle\Service\ParameterToEntityMapBuilder;
+use Paysera\Component\Serializer\Factory\ContextAwareNormalizerFactory;
+
+class RestListenerLocaleTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var MockInterface|GetResponseEvent
+     */
+    private $responseEvent;
+
+    public function setUp()
+    {
+        $this->responseEvent = Mockery::mock(GetResponseEvent::class);
+    }
+
+    /**
+     * @param array $acceptedLocales
+     * @param string $expected
+     * @param string|null $acceptLanguage
+     *
+     * @dataProvider dataProviderForTestLocaleIsBeingSet
+     */
+    public function testCorrectLocaleIsBeingSet(array $acceptedLocales, $expected, $acceptLanguage = null)
+    {
+        $restListener = $this->createRestListener($acceptedLocales);
+
+        $request = new Request();
+
+        if ($acceptLanguage !== null) {
+            $request->headers->set('Accept-Language', $acceptLanguage);
+        }
+
+        $this->responseEvent->shouldReceive('getRequest')->andReturn($request);
+
+        $restListener->onKernelRequest($this->responseEvent);
+
+        $this->assertEquals($expected, $request->getLocale());
+    }
+
+    /**
+     * @return array
+     */
+    public function dataProviderForTestLocaleIsBeingSet()
+    {
+        return [
+            'select requested locale' => [
+                ['lt', 'ru'],
+                'lt',
+                'lt'
+            ],
+            'select locale with the highest weight' => [
+                [],
+                'zh',
```

Comment:
Right here we should leave the request locale as it was before that. Otherwise it's backwards incompatible change that could set unexpected locales without whitelisting them